src/averageAges.js

```diff
@@ -52,7 +71,22 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = !onlyWithSon
+    ? people.filter(person => people.some(x => person.mother === x.name))
+    : people.filter(person => {
+      return person.sex === 'm'
+      && people.some(x => person.mother === x.name);
+    });
```

Comment:
And same here. You can use method .filter only once

Reply:
all done)